src/subworkflows/subworkflow.js

```diff
@@ -108,6 +108,11 @@ export class Subworkflow extends BaseSubworkflow {
         // reset units if application name changes
         if (this.application.name !== application.name) this.setUnits([]);
         this._application = application;
+        this.units.forEach((unit) => {
+            if (typeof unit.setApplication === "function") {
+                unit.setApplication(application, false);
+            }
+        });
```

Comment:
This needs some more thinking:

1. This needs to take into account the above logic regarding the application name change

```js
if (this.application.name !== application.name) {
    this.setUnits([]);
} else {
    this.units.filter(unit => typeof unit.setApplication === "function").foreach(unit => unit.setApplication(application, true))
}
```

Reply:
Is the only concern here to prevent running update logic after units have been cleared on name change?